backend/api/EventHandlers/MissionEventHandler.cs

```diff
@@ -149,8 +149,24 @@ private async void OnRobotAvailable(object? sender, RobotAvailableEventArgs e)
             }
 
             MissionRun? missionRun;
-            if (robot.MissionQueueFrozen) { missionRun = await MissionService.ReadNextScheduledEmergencyMissionRun(robot.Id); }
-            else { missionRun = await MissionService.ReadNextScheduledMissionRun(robot.Id); }
+
+            missionRun = await MissionService.ReadNextScheduledLocalizationMissionRun(robot.Id);
+
+            if (missionRun != null)
```

Comment:
Could this be changed to a guard statement?

Reply:
The guard statement being if(missionRun is not null), dont see upside ?

Reply:
Done.